core/services/gateway/handlers/webapicapabilities/handler.go

```diff
@@ -0,0 +1,156 @@
+package webapicapabilities
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/config"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/common"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/network"
+)
+
+const (
+	// NOTE: more methods will go here. HTTP trigger/action/target; etc.
+	MethodWebAPITarget = "web_api_target"
+)
+
+type handler struct {
+	don             handlers.DON
+	lggr            logger.Logger
+	httpClient      network.HTTPClient
+	nodeRateLimiter *common.RateLimiter
+}
+
+type HandlerConfig struct {
+	NodeRateLimiter common.RateLimiterConfig `json:"nodeRateLimiter"`
+}
+
+var _ handlers.Handler = (*handler)(nil)
+
+func NewHandler(handlerConfig json.RawMessage, donConfig *config.DONConfig, don handlers.DON, httpClient network.HTTPClient, lggr logger.Logger) (*handler, error) {
+	var cfg HandlerConfig
+	err := json.Unmarshal(handlerConfig, &cfg)
+	if err != nil {
+		return nil, err
+	}
+	nodeRateLimiter, err := common.NewRateLimiter(cfg.NodeRateLimiter)
+	if err != nil {
+		return nil, err
+	}
+
+	return &handler{
+		don:             don,
+		lggr:            lggr.Named("WebAPIHandler." + donConfig.DonId),
+		httpClient:      httpClient,
+		nodeRateLimiter: nodeRateLimiter,
+	}, nil
+}
+
+func (h *handler) HandleUserMessage(ctx context.Context, msg *api.Message, callbackCh chan<- handlers.UserCallbackPayload) error {
+	return nil
+}
+
+// sendHTTPMessageToClient is an outgoing message from the gateway to external endpoints
+// returns message to be sent back to the capability node
+func (h *handler) sendHTTPMessageToClient(ctx context.Context, req network.HTTPRequest, msg *api.Message) (*api.Message, error) {
+	var payload TargetResponsePayload
+	resp, err := h.httpClient.Send(ctx, req)
+	if err != nil {
+		return nil, err
+	} else {
+		payload = TargetResponsePayload{
+			ExecutionError: false,
+			StatusCode:     uint16(resp.StatusCode),
+			Headers:        resp.Headers,
+			Body:           resp.Body,
+		}
+	}
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return nil, err
+	}
+	return &api.Message{
+		Body: api.MessageBody{
+			MessageId: msg.Body.MessageId,
+			Method:    msg.Body.Method,
+			DonId:     msg.Body.DonId,
+			Payload:   payloadBytes,
+		},
+	}, nil
+}
+
+func (h *handler) handleWebAPITargetMessage(ctx context.Context, msg *api.Message, nodeAddr string) error {
+	h.lggr.Debugw("handling web api target message", "messageId", msg.Body.MessageId, "nodeAddr", nodeAddr)
+	if !h.nodeRateLimiter.Allow(nodeAddr) {
+		return fmt.Errorf("rate limit exceeded for node %s", nodeAddr)
+	}
+	var targetPayload TargetRequestPayload
```

Comment:
Are you expecting there to be any different between TargetRequestPayload and network.HTTPRequest?

If so I'm wondering if we could just use network.HTTPRequest throughout? This will be beneficial for the Compute Fetch API which will effectively be sending the same kinds of messages anyway.

Reply:
Interesting. They are essentially the same shapes for now so I'm open to combining them. One example I can think of is additional fields in `TargetRequestPayload` (or `WebAPIRequestPayload` that can be used for both target and action) that specify some aspects about the gateway interaction (e.g. gateway address to use for outgoing http message or even dynamic delivery mode, which is currently a workflow-level setting)

Reply:
If you think they might need to diverge, let's probably leave it as is for now :)